### PR TITLE
Include all resources in a build target's sources list

### PR DESCRIPTION
**Pull request by matthewbastien**
Companion PR to swiftlang/swift-package-manager#8107. This will allow SourceKit-LSP to be able to find Swift DocC documentation catalogs within SwiftPM build targets.

### CODE REVIEW

File: Sources/BuildSystemIntegration/SwiftPMBuildSystem.swift [hunk 1]
```diff
@@ -565,6 +565,27 @@ package actor SwiftPMBuildSystem: BuiltInBuildSystem {
...
+          generated: false,
+        )
+      }
+      sources += swiftPMTarget.others.map {
+        SourceItem(
+          uri: DocumentURI($0),
+          kind: $0.isDirectory ? .directory : .file,
+          generated: false,
+        )
+      }
```
Comment (ahoppen) [inline Sources/BuildSystemIntegration/SwiftPMBuildSystem.swift:588]: The maps are all the same. Could we have only one copy of them as follows?

```suggestion
      sources += (swiftPMTarget.resources + swiftPMTarget.ignored + swiftPMTarget.others).map {
        SourceItem(
          uri: DocumentURI($0),
          kind: $0.isDirectory ? .directory : .file,
          generated: false,
        )
      }
```

Comment (matthewbastien) [reply Sources/BuildSystemIntegration/SwiftPMBuildSystem.swift]: Good call!

**Comment by matthewbastien** [comment]
@swift-ci please test

**Review by ahoppen — APPROVED** [review]

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Pull request merged**